java/arcs/android/devTools/DevWebSocket.kt

```diff
@@ -0,0 +1,51 @@
+package arcs.android.devTools
+
+import fi.iki.elonen.NanoHTTPD
+import fi.iki.elonen.NanoWSD
+import java.io.IOException
+
+class DevWebSocket : NanoWSD(12345) {
+    var wsd: WsdSocket? = null
+    protected override fun openWebSocket(ihttpSession: NanoHTTPD.IHTTPSession?): WebSocket {
+        wsd = WsdSocket(ihttpSession)
+        return wsd ?: WsdSocket(ihttpSession)
+    }
+
+    fun send(msg: String) {
+        wsd?.send(msg)
+    }
+
+    public class WsdSocket(
+        handshakeRequest: NanoHTTPD.IHTTPSession?
+    ) : WebSocket(handshakeRequest) {
+        private val PING_PAYLOAD = "1337DEADBEEFC001".toByteArray()
+
+        protected override fun onOpen() {
+            try {
+                send("Socket Open")
+                ping(PING_PAYLOAD)
+            } catch (e: IOException) {
+            }
+        }
+
+        protected override fun onClose(
+            code: WebSocketFrame.CloseCode?,
+            reason: String,
+            initiatedByRemote: Boolean
+        ) {
+        }
+
+        // override onOpen, onClose, onPong and onException methods
+        protected override fun onMessage(webSocketFrame: WebSocketFrame) {
+            try {
+                send(webSocketFrame.getTextPayload().toString() + " to you")
+            } catch (e: IOException) { // handle
+            }
+        }
+
+        protected override fun onPong(pong: WebSocketFrame?) {
```

Comment:
Add a comment that `NanoWSD` requires explicit ping handling and if not done the connection is terminated.